autofit/non_linear/search/mle/bfgs/search.py

```diff
@@ -0,0 +1,338 @@
+from typing import Optional
+
+from autoconf import cached_property
+from autofit.database.sqlalchemy_ import sa
+
+from autofit.mapper.prior_model.abstract import AbstractPriorModel
+from autofit.non_linear.search.mle.abstract_mle import AbstractMLE
+from autofit.non_linear.analysis import Analysis
+from autofit.non_linear.fitness import Fitness
+from autofit.non_linear.initializer import AbstractInitializer
+from autofit.non_linear.samples.sample import Sample
+from autofit.non_linear.samples.samples import Samples
+
+import copy
+from scipy import optimize
+import numpy as np
+
+
+class AbstractBFGS(AbstractMLE):
+    __identifier_fields__ = ()

```

Comment:
Probably best not to define an empty __identifier_fields__ - I think as it stands it would mean attributes would be ignored when creating an identifier for an AbstractBFGS meaning all AbstractBFGS runs of the same type would have the same identifier